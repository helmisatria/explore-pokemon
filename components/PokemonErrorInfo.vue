<template>
  <div class="error-information p-6 flex flex-col justify-center space-y-2">
    <p v-if="onlineStatus === 'offline'">You're offline, please check your connection</p>
    <p v-else>Something went wrong, please try again</p>
    <button class="bg-green-500 rounded px-3 py-2 text-white" @click="reload">Try Again</button>
  </div>
</template>

<script>
import { mapState } from 'pinia'
import { usePokemonStore } from '~/store'

export default {
  name: 'PokemonErrorInfo',
  computed: {
    ...mapState(usePokemonStore, ['onlineStatus']),
  },
  watch: {
    onlineStatus(newStatus, previousStatus) {
      if (previousStatus !== 'online' && newStatus === 'online') {
        this.reload()
      }
    },
  },
  methods: {
    reload() {
      this.$emit('reload')
    },
  },
}
</script>
