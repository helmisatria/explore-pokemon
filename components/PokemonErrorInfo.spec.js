import { createLocalVue, shallowMount } from '@vue/test-utils'
import { PiniaVuePlugin } from 'pinia'
import { createTestingPinia } from '@pinia/testing'
import PokemonErrorInfo from './PokemonErrorInfo.vue'

describe('Component: PokemonErrorInfo', () => {
  let wrapper

  const localVue = createLocalVue()
  localVue.use(PiniaVuePlugin)

  const generateWrapper = () => {
    return shallowMount(PokemonErrorInfo, {
      localVue,
      pinia: createTestingPinia(),
    })
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should mount component properly', () => {
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('.error-information > p').text()).toBe(
      'Something went wrong, please try again'
    )
  })

  it('should emit reload, given user click the reload button', () => {
    wrapper.find('button').trigger('click')
    expect(wrapper.emitted('reload')).toBeTruthy()
  })
})
