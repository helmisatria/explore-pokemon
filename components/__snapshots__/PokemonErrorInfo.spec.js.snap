// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PokemonErrorInfo should mount component properly 1`] = `
<div
  class="error-information p-6 flex flex-col justify-center space-y-2"
>
  <p>
    Something went wrong, please try again
  </p>
   
  <button
    class="bg-green-500 rounded px-3 py-2 text-white"
  >
    Reload
  </button>
</div>
`;
